### PR TITLE
fix(helm): fix queryScheduler servicemonitor

### PR DESCRIPTION
**Pull request by rlex**
**What this PR does / why we need it**:
Loki Query Scheduler had wrong targetPort for metrics port, which caused serviceMonitor to miss it.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by JStickler** [comment]
Please also bump the version number in `production/helm/loki/README.md.`

**Comment by rlex** [comment]
done

**Comment by rlex** [comment]
Any chance it will be merged? Along with #12766

**Review by MichelHollands — APPROVED** [review]

File: production/helm/loki/templates/query-scheduler/service-query-scheduler.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ spec:
   ports:
     - name: http-metrics
       port: 3100
-      targetPort: http
+      targetPort: http-metrics
```
Comment (yoyosir) [inline production/helm/loki/templates/query-scheduler/service-query-scheduler.yaml:24]: Do we have the same issue for service-query-frontend-headless.yaml?

Comment (rlex) [reply production/helm/loki/templates/query-scheduler/service-query-scheduler.yaml:24]: https://github.com/grafana/loki/commit/1ab9d271c354caf0ba589691e6477fb9a19039f0 looks like it was fixed just recently

**Pull request merged**